src/Plugin/WebformElement/NetsEasyPaymentElement.php

```diff
@@ -176,26 +185,12 @@ public function alterForm(array &$element, array &$form, FormStateInterface $for
    *   Returns validation results.
    */
   public static function validatePayment(array &$element, FormStateInterface $formState): mixed {
-    $paymentHelper = \Drupal::service('Drupal\os2forms_payment\Helper\PaymentHelper');
-    $paymentElementClass = get_called_class();
-
-    $paymentId = $formState->getValue($paymentElementClass::PAYMENT_REFERENCE_NAME);
-    if (!$paymentId) {
-      return $formState->setError(
-        $element,
-        t('The form could not be submitted. Please try again.')
-      );
-    }
-
-    $endpoint = $paymentHelper->getPaymentEndpoint() . $paymentId;
+    $paymentHelper = \Drupal::service(PaymentHelper::class);
 
-    $paymentValidated = $paymentHelper->validatePayment($endpoint);
+    $paymentHelper->validatePayment($element, $formState);
 
-    if (!$paymentValidated) {
-      return $formState->setError(
-        $element,
-        t('The payment could not be validated. Please try again.')
-      );
+    if ($formState->hasAnyErrors()) {
+      return FALSE;
```

Comment:
Is the return value from the validation function used anywhere? It's my understanding that just setting errors on the form object is all that's needed, i.e. the validation method should have side-effects.

Reply:
Right. It was not needed.